src/models.js

```diff
@@ -110,6 +116,11 @@ export class TxHistory {
     return this.version === constants.NANO_CONTRACTS_VERSION;
   }
 
+  hasFirstBlock() {
+    // not null, not undefined and not string empty
+    return this.firstBlock != null && this.firstBlock.trim() !== '';
```

Comment:
todo: It's important that this validation is made with `!==` instead of `!=`.
The behavior of comparing something with `!=` null can be hard to predict.
```suggestion
    return this.firstBlock !== null && this.firstBlock.trim() !== '';
```

Reply:
The use of `!= null` was intentional. What it does check of the value is not null and not undefined.

Reply:
As we've talked, I just feel that the behavior of the `==` / `!=` operators [is a bit hard to predict](https://blog.webdevsimplified.com/2020-08/==-vs-===/#:~:text=Conclusion,equals%20is%20the%20preferred%20method.).
But if this has been well tested I see no problem in keeping it.

Reply:
This is the precise case it is good to use, indeed:
![image](https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/6dd2ca07-88b3-486a-b80b-7608d7ad412b)